src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -2687,7 +2687,7 @@ ImmutableArray<IOperation> getOperationsToAnalyzeWithStackGuard(ImmutableArray<I
                 {
                     return GetOperationsToAnalyze(operationBlocksToAnalyze);
                 }
-                catch (Exception ex) when (ex is InsufficientExecutionStackException || FatalError.ReportAndCatchUnlessCanceled(ex))
+                catch (Exception ex) when (ex is InsufficientExecutionStackException || FatalError.ReportAndCatchUnlessCanceled(ex, cancellationToken))
```

Comment:
This one could result in a user-visible AD0002 for IDE scenarios, but the associated stack trace would allow us to resolve it.